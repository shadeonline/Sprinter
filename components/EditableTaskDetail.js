import React, { useState } from 'react';
import { View, TextInput, TouchableOpacity, StyleSheet, Text } from 'react-native';

export default EditableTaskDetail = ({ task, editedTask, setEditedTask, handleSaveChanges, setIsEditing }) => {
    return (
        <View style={styles.editableForm}>
            <Text style={styles.modalText}>Title:</Text>
            <TextInput
                style={styles.editableText}
                value={editedTask.taskTitle}
                onChangeText={(text) =>
                    setEditedTask({ ...editedTask, taskTitle: text })
                }
            />
            <Text style={styles.modalText}>Storypoint:</Text>
            <TextInput
                style={styles.editableText}
                value={editedTask.storyPoint}
                onChangeText={(text) =>
                    setEditedTask({ ...editedTask, storyPoint: text })
                }
            />
            <Text style={styles.modalText}>Description:</Text>
            <TextInput
                style={[styles.editableText, styles.descriptionInput]}
                multiline={true}
                numberOfLines={4}
                value={editedTask.taskDescription}
                onChangeText={(text) =>
                    setEditedTask({ ...editedTask, taskDescription: text })
                }
            />
            <Text style={styles.modalText}>Deadline:</Text>
            <View style={styles.deadlineContainer}>
                <TextInput
                    style={styles.deadlineInput}
                    placeholder="dd"
                    value={editedTask.deadline.split('/')[0]}
                    onChangeText={(text) =>
                        setEditedTask({
                            ...editedTask,
                            deadline: `${text}/${editedTask.deadline.split('/')[1]}/${editedTask.deadline.split('/')[2]}`,
                        })
                    }
                />
                <Text style={styles.deadlineSeparator}>/</Text>
                <TextInput
                    style={styles.deadlineInput}
                    placeholder="mm"
                    value={editedTask.deadline.split('/')[1]}
                    onChangeText={(text) =>
                        setEditedTask({
                            ...editedTask,
                            deadline: `${editedTask.deadline.split('/')[0]}/${text}/${editedTask.deadline.split('/')[2]}`,
                        })
                    }
                />
                <Text style={styles.deadlineSeparator}>/</Text>
                <TextInput
                    style={styles.deadlineInput}
                    placeholder="yy"
                    value={editedTask.deadline.split('/')[2]}
                    onChangeText={(text) =>
                        setEditedTask({
                            ...editedTask,
                            deadline: `${editedTask.deadline.split('/')[0]}/${editedTask.deadline.split('/')[1]}/${text}`,
                        })
                    }
                />
            </View>

            <View style={styles.buttonContainer}>
                {/* CANCEL BUTTON */}
                <TouchableOpacity
                    style={styles.cancelButton}
                    onPress={() => {
                        setIsEditing(false);
                        setEditedTask(task);
                    }}
                >
                    <Text style={styles.cancelButtonText}>Cancel</Text>
                </TouchableOpacity>
                
                {/* SAVE BUTTON */}
                <TouchableOpacity
                    style={styles.saveButton}
                    onPress={handleSaveChanges}
                >
                    <Text style={styles.saveButtonText}>Save Changes</Text>
                </TouchableOpacity>
            </View>
        </View>
    );
};
const styles = StyleSheet.create({
    modalContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.5)',
    },
    modalContent: {
        backgroundColor: 'white',
        padding: 16,
        borderRadius: 8,
        width: '90%',
        alignItems: 'center',
        position: 'relative',
    },
    modalTitle: {
        fontSize: 20,
        fontWeight: 'bold',
        marginBottom: 8,
    },
    modalText: {
        fontSize: 14,
        color: 'gray',
        marginBottom: 8,
        marginTop:10,
        fontWeight: 'bold',
    },
    editableForm: {
        width: '100%',
    },
    modalDetails: {
        width: '100%',
    },
    editableText: {
        fontSize: 14,
        color: 'gray',
        marginBottom: 8,
        borderWidth: 1,
        borderColor: '#E1E3E8',
        borderRadius: 4,
        padding: 8,
        width: '100%',
    },
    descriptionInput: {
        height: 100,
        textAlignVertical: 'top',
    },
    closeButton: {
        position: 'absolute',
        top: 10,
        right: 10,
        backgroundColor: 'transparent',
        padding: 10,
        zIndex: 1,
    },
    closeButtonText: {
        color: 'black',
        fontWeight: 'bold',
        fontSize: 18,
    },
    buttonContainer: {
        flexDirection: 'row',
        marginTop: 16,
        justifyContent: 'center', // Center the buttons horizontally.
    },
    editButton: {
        flex: 1,
        backgroundColor: '#0052CC',
        padding: 10,
        borderRadius: 4,
        marginRight: 8,
    },
    editButtonText: {
        color: 'white',
        fontWeight: 'bold',
        textAlign: 'center',
    },
    deleteButton: {
        flex: 1,
        backgroundColor: '#5386e4',
        padding: 10,
        borderRadius: 4,
    },
    deleteButtonText: {
        color: 'white',
        fontWeight: 'bold',
        textAlign: 'center',
    },
    saveButton: {
        flex: 1,
        margin: 5,
        backgroundColor: 'green',
        padding: 10,
        borderRadius: 4,
    },
    saveButtonText: {
        color: 'white',
        fontWeight: 'bold',
        textAlign: 'center',
    },
    cancelButton: {
        flex: 1,
        margin: 5,
        backgroundColor: '#DD2C00',
        padding: 10,
        borderRadius: 4,
    },
    cancelButtonText: {

        color: 'white',
        fontWeight: 'bold',
        textAlign: 'center',
    },
    deadlineContainer: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    deadlineInput: {
        flex: 1,
        fontSize: 14,
        color: 'gray',
        marginBottom: 8,
        borderWidth: 1,
        borderColor: '#E1E3E8',
        borderRadius: 4,
        padding: 8,
    },
    deadlineSeparator: {
        fontSize: 14,
        color: 'gray',
        marginHorizontal: 4,
    },
});