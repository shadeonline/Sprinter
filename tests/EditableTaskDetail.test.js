import React from 'react';
import renderer from 'react-test-renderer';
import EditableTaskDetail from '../components/EditableTaskDetail';

describe('EditableTaskDetail', () => {
  it('matches snapshot', () => {
    const task = {
      taskTitle: 'Sample Task',
      storyPoint: '3',
      taskDescription: 'This is a sample task description.',
      deadline: '30/09/23',
    };
    const editedTask = {
      taskTitle: 'Edited Task',
      storyPoint: '5',
      taskDescription: 'This is the edited task description.',
      deadline: '15/10/23',
    };
    const setEditedTask = jest.fn();
    const handleSaveChanges = jest.fn();
    const setIsEditing = jest.fn();

    const tree = renderer
      .create(
        <EditableTaskDetail
          task={task}
          editedTask={editedTask}
          setEditedTask={setEditedTask}
          handleSaveChanges={handleSaveChanges}
          setIsEditing={setIsEditing}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
