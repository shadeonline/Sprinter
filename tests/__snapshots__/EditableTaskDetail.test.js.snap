// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTaskDetail matches snapshot 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginBottom": 8,
        "marginTop": 10,
      }
    }
  >
    Title:
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      {
        "borderColor": "#E1E3E8",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "gray",
        "fontSize": 14,
        "marginBottom": 8,
        "padding": 8,
        "width": "100%",
      }
    }
    value="Edited Task"
  />
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginBottom": 8,
        "marginTop": 10,
      }
    }
  >
    Storypoint:
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      {
        "borderColor": "#E1E3E8",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "gray",
        "fontSize": 14,
        "marginBottom": 8,
        "padding": 8,
        "width": "100%",
      }
    }
    value="5"
  />
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginBottom": 8,
        "marginTop": 10,
      }
    }
  >
    Description:
  </Text>
  <TextInput
    multiline={true}
    numberOfLines={4}
    onChangeText={[Function]}
    style={
      [
        {
          "borderColor": "#E1E3E8",
          "borderRadius": 4,
          "borderWidth": 1,
          "color": "gray",
          "fontSize": 14,
          "marginBottom": 8,
          "padding": 8,
          "width": "100%",
        },
        {
          "height": 100,
          "textAlignVertical": "top",
        },
      ]
    }
    value="This is the edited task description."
  />
  <Text
    style={
      {
        "color": "gray",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginBottom": 8,
        "marginTop": 10,
      }
    }
  >
    Deadline:
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="dd"
      style={
        {
          "borderColor": "#E1E3E8",
          "borderRadius": 4,
          "borderWidth": 1,
          "color": "gray",
          "flex": 1,
          "fontSize": 14,
          "marginBottom": 8,
          "padding": 8,
        }
      }
      value="15"
    />
    <Text
      style={
        {
          "color": "gray",
          "fontSize": 14,
          "marginHorizontal": 4,
        }
      }
    >
      /
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="mm"
      style={
        {
          "borderColor": "#E1E3E8",
          "borderRadius": 4,
          "borderWidth": 1,
          "color": "gray",
          "flex": 1,
          "fontSize": 14,
          "marginBottom": 8,
          "padding": 8,
        }
      }
      value="10"
    />
    <Text
      style={
        {
          "color": "gray",
          "fontSize": 14,
          "marginHorizontal": 4,
        }
      }
    >
      /
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="yy"
      style={
        {
          "borderColor": "#E1E3E8",
          "borderRadius": 4,
          "borderWidth": 1,
          "color": "gray",
          "flex": 1,
          "fontSize": 14,
          "marginBottom": 8,
          "padding": 8,
        }
      }
      value="23"
    />
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#DD2C00",
          "borderRadius": 4,
          "flex": 1,
          "margin": 5,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Cancel
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "green",
          "borderRadius": 4,
          "flex": 1,
          "margin": 5,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Save Changes
      </Text>
    </View>
  </View>
</View>
`;
